Source/controls/controller_motion.cpp

```diff
@@ -154,15 +155,15 @@ AxisDirection GetLeftStickOrDpadDirection(bool allowDpad)
 
 	AxisDirection result { AxisDirectionX_NONE, AxisDirectionY_NONE };
 
-	if (stickY >= 0.5 || (allowDpad && IsControllerButtonPressed(ControllerButton_BUTTON_DPAD_UP))) {
+	if (stickY >= 0.5 || (allowDpad && IsControllerButtonPressed(ControllerButton_BUTTON_DPAD_UP)) || virtualGamepad.IsControllerButtonPressed(ControllerButton_BUTTON_DPAD_UP)) {

```

Comment:
Perhaps the virtual gamepad check should be moved into `IsControllerButtonPressed`?

Reply:
Perhaps, but that would cause the virtual d-pad to be disabled along with the physical d-pad via the INI setting, which doesn't make much sense to me.